zeebe/engine/src/main/java/io/camunda/zeebe/engine/state/metrics/PersistedUsageMetrics.java

```diff
@@ -89,4 +106,11 @@ public PersistedUsageMetrics recordEDI(final String tenantId) {
     setTenantEDIMap(tenantEDIMap);
     return this;
   }
+
+  public PersistedUsageMetrics recordTU(final String tenantId, final String assignee) {
+    final var tenantTUMap = getTenantTUMap();
+    tenantTUMap.computeIfAbsent(tenantId, k -> new HashSet<>()).add(assignee);
+    setTenantTUMap(tenantTUMap);
+    return this;
```

Comment:
🔧 We could add a condition to only update state when the assignee is not in the set yet. I think it will perform better than always trying to add it.

Reply:
Good point with the `Set.add`, however we still call the `setTenantTUMap`.

We could do that part conditionally, something like this:

```suggestion
    final var tenantTUMap = getTenantTUMap();
    boolean added = tenantTUMap.computeIfAbsent(tenantId, k -> new HashSet<>()).add(assignee);
    if (added) setTenantTUMap(tenantTUMap);
    return this;
```

That should have some advantage right? Because we don't update the state unnecessarily.